iroh/src/protocol.rs

```diff
@@ -300,12 +302,28 @@ impl RouterBuilder {
                                     break;
                                 }
                             }
-                            Some(Ok(Err(inner))) => {
-                                debug!("Task errored: {inner:?}");
+                            Some(Ok(Some(()))) => {
+                                debug!("Task finished");
+                            }
+                            Some(Ok(None)) => {
+                                debug!("Task cancelled");
                             }
                             _ => {}
                         }
                     },
+
+                    // handle incoming p2p connections.
+                    incoming = endpoint.accept() => {
+                        let Some(incoming) = incoming else {
+                            break;
+                        };
+
+                        let protocols = protocols.clone();
+                        let token = cancel_token.child_token();
+                        join_set.spawn(async move {
+                            token.run_until_cancelled(handle_connection(incoming, protocols)).await
+                        });
```

Comment:
```suggestion
                        join_set.spawn(token.run_until_cancelled(handle_connection(incoming, protocols)));
```
I'm a sucker for pointfree style. That said, this might not compile. And it might not actually read better to others.

Reply:
doesn't compile..